ext/curl/curl_file.stub.php

```diff
@@ -1,9 +1,19 @@
 <?php
 
-/** @generate-function-entries */
+/**
+ * @generate-function-entries
+ * @generate-property-declarations
```

Comment:
In terms of general approach, the angle I would go for is `@generate-class-entries`, where we generate the full code for registering a class.

Reply:
And I'd also put it inside the main arginfo.h file, rather than a separate one.

Reply:
Putting in the `_arginfo.h` works for me, and I'm also ok with generating the class initialization code.

My only concern with generating the whole code is that we should know the PHP class - `zend_class_entry` relation if we also want to generate the code for adding parents/interfaces. Or should we require class entries to follow a naming convention?

Reply:
An idea came to my mind: we could require the parent/interface class entries to be passed as arguments to the generator function.

Reply:
I tried to implement everything for making auto-generating class entries possible. Unfortunately, I'm getting a segmentation fault when the phar is being generated for the tests, so I'm wondering if there is something particularly wrong with my approach. In the meanwhile, I converted the function to a macro as I wasn't sure that class registration would have worked via using a function.